src/documentalist.ts

```diff
@@ -36,49 +43,110 @@ const RESERVED_WORDS = [
     "import",
 ];
 
-export interface IBlock {
-    content: string;
-    metadata: any;
-    renderedContent: StringOrTag[];
-}
-
-export interface IApi {
-    /** Process the given globs and emit all the data. */
-    documentGlobs: (...filesGlobs: string[]) => IDocumentalistData;
+export interface IApi<T> {
+    /**
+     * Finds all files matching the provided variadic glob expressions and then
+     * runs `documentFiles` with them, emitting all the documentation data.
+     *
+     * @see documentFiles
+     */
+    documentGlobs: (...filesGlobs: string[]) => T;
 
-    /** Process the given list of files and emit all the data. */
-    documentFiles: (files: IFile[]) => IDocumentalistData;
+    /**
+     * Iterates over all plugins, passing all matching files to each in turn.
+     * The resulting output for each plugin is combined into the resulting
+     * documentation object.
+     *
+     * The documentation object has a composite type of all the plugin data
+     * types.
+     */
+    documentFiles: (files: IFile[]) => T;
 
     /**
-     * Render a block of content by extracting metadata front matter and
-     * splitting text content into rendered HTML strings and `{ tag, value }` objects.
+     * Render a block of content by extracting metadata (YAML front matter) and
+     * splitting text content into markdown-rendered HTML strings and `{ tag,
+     * value }` objects.
+     *
+     * To prevent special strings like "@include" from being parsed, a reserved
+     * tag words array may be provided, in which case the line will be left as
+     * is.
      */
     renderBlock: (blockContent: string, reservedTagWords?: string[]) => IBlock;
 
-    /** Use a plugin to process the files matching the pattern. */
-    use: (pattern: RegExp, plugin: IPlugin<any>) => IApi;
+    /**
+     * Adds the plugin to Documentalist. Returns a new instance of Documentalist
+     * with a template type that includes the data from the plugin. This way the
+     * `documentFiles` and `documentGlobs` methods will return an object that is
+     * already typed to include the plugin output.
+     *
+     * The Plugin is applied to all files whose absolute path matches the
+     * supplied pattern.
+     *
+     * @param pattern - A regexp pattern or a file extension string like "js"
```

Comment:
i prefer the full extension like ".js", makes it more obvious what it's for. if you don't insert the period in regexp then both options will work the same